### PR TITLE
Ruby on Rails Capstone: Budget App

### PR DESCRIPTION
**Pull request by yigitm**
#### Design
- Followed the given design:  [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)
- Mentioned in Readme file due to "Creative Commons license of the design"

#### Interactions
- Splash screen
  - A simple page with the name of the Budget App, and links to the sign-up and log-in pages.

- Sign up and login pages:
  - Users are able to register in the app with their full name, email, and password (all mandatory).
  - Users can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in 

- Home page (categories page) - group model
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page - payment model
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page, the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new payment" page
  - The user fills out a form to create a new transaction with name & amount (both are mandatory)
  -  Current category is mandatory in the form
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements

- Created unit and integration tests for login, signup and two main model group & payment

#### Technical requirements

- Postgres is the database
- Devise is used for authentication
- User inputs are validated
- View template engine is: (.erb)
- The project is deployed and accessible online with Heroku
- Your database schema reflects the requested structure in ERD

Thank you for your time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,108 @@
...
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Microverse
+- Open source
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (ahmetbozaci) [inline README.md]: - [x] Please fix the broken link in the license.
- [x] Also, you should add a creative common license. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=The%20Creative%20Commons%20license%20of%20the%20design%20requires%20that%20you%20give%20appropriate%20credit%20to%20the%20author.%20Therefore%2C%20you%20must%20do%20it%20in%20the%20README%20of%20your%20project.).

File: README.md [hunk 2]
```diff
@@ -1,24 +1,108 @@
...
+Feel free to check the [issues page](https://github.com/yigitm/budget-app.git/issues).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Microverse
+- Open source
```
Comment (ahmetbozaci) [inline README.md]: - [x] You should add a design link.  [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=Original%20design%20idea%20by%20Gregoire%20Vella%20on%20Behance.).

File: app/views/payments/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      <h1 class="text-7xl text-center my-20">No payments yet</h1>
+    </div>
+  <% end %>
+</div>
+
+<div class="w-full bg-lime-400 p-5 mt-5 absolute bottom-0">
+  <div class="border border-solid border-7 border-white w-50 rounded-lg">
+    <%= link_to 'New payment', new_group_payment_path, class: "py-5 text-white text-center block font-medium mt-5 mb-3 w-70" %>
+  </div>
+</div>
```
Comment (ahmetbozaci) [inline app/views/payments/index.html.erb]: - [x] The list of payments should present ordered by the most recent. Please fix it.
image
- [x] I like to go back to the category page from this page. Please add a back button that navigates to the home page(category page).

___
### OPTIONAL

- [ ] Also you can see after creating many payments the button overlap with payments. You can fix it.
image

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,12 @@
...
+  devise_for :users
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
   # root "articles#index"
+  root to: "groups#index"
+
+  resources :groups do
+    resources :payments
+  end
```
Comment (ahmetbozaci) [inline config/routes.rb:11]: - [ ] Please add only necessary routes. You can do it by using `only`.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @yigitm  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- Nice to add some flash messages. ✔️ 
- Deployed Application ✔️ 
- Nice video Presentation ✔️ 


___


### ♻️ Request changes ♻️

- [x] _Check the comments under the review._
- [x] Please follow the design guidelines therefore you should change the font family. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=Project%20requirements-,Design,Layout%3A%20composition%20and%20space%20between%20elements.,-NOTE%3A%20In%20these)
-You are using a different font family
image

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: README.md (hunk 1 above)
Comment (yigitm) [reply README.md]: Thank you. I added the fonts.

File: README.md (hunk 2 above)
Comment (yigitm) [reply README.md]: Thank you. I added the link to "Acknowledgments".

File: README.md (hunk 1 above)
Comment (yigitm) [reply README.md]: and creative common license with fixed MIT link, too.

File: app/views/payments/index.html.erb (hunk 3 above)
Comment (yigitm) [reply app/views/payments/index.html.erb]: I will fix it soon. Thanks.

File: config/routes.rb (hunk 4 above)
Comment (yigitm) [reply config/routes.rb:11]: I will do it soon. Thanks.

File: README.md (hunk 1 above)
Comment (rloterh) [reply README.md]: This issue is well resolved! ✔️

File: README.md (hunk 2 above)
Comment (rloterh) [reply README.md]: This issue is well resolved! ✔️

**Review by Deepakdanger — APPROVED** [review]
Hi @yigitm, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**